web/app/widgets/button/button.tpl.html

```diff
@@ -1,4 +1,11 @@
-<div class="box-content switch" ng-style="{ 'background-color': vm.background, color: vm.foreground }" ng-click="vm.sendCommand()">
+<style>
+    .button-dummy .box-content.dummy {
+        background-color: {{vm.background}};
+        color: {{vm.foreground}};
+    }
+</style>
```

Comment:
Adding inline styles in `<style>`elements is discouraged unless in specific places like modals; furthermore, this won't work if you add several buttons - there is no concept of "scoped" or "virtual" DOM here so any added styling is global.

Styles for widgets should now be added in their associated .scss file if necessary, here the currently empty `button.scss`, and then processed with the `gulp sass` command.

I think the contents of the dummy widget template should rather be merged ("clerverly" or as a separate `<div ng-if="widget.show_item_value">`) into the button template instead of calling the `widget-dummy` directive. Of course the buttons should remain looking the same as before if the value is not shown.

Reply:
Yeah I thought about that also but since the button now can mimic the dummy, whenever there's any change in dummy will have to be done for the button template as well. With what I did, changes to dummy template will require no change in the button. 

Also if we want to mimic the dummy and still be able to change bg color, we'll have to implement that to the dummy itself. I can do that, let me know.

Reply:
Since this style is in that div, it will be scoped to that parent div. But with the new changes I will push, this will be irrelevant now. Im using the actual same html/tpl as dummy